### PR TITLE
Share survey story 4

### PR DESCRIPTION
**Pull request by creativeappdevamit**
# What and why?
https://trello.com/c/3oNFsYdm/482-s34-user-account-creation-share-surveys-holds-account-story-4
# How to test?

# Trello
https://trello.com/c/3oNFsYdm/482-s34-user-account-creation-share-surveys-holds-account-story-4

### CODE REVIEW

File: test/party_client.py [hunk 1]
```diff
@@ -366,3 +366,16 @@ def post_share_surveys_fail(self, payload, expected_status=400):
     def delete_share_surveys(self, expected_status=204):
         response = self.client.delete(f'/party-api/v1/batch/pending-shares', headers=self.auth_headers)
         self.assertStatus(response, expected_status)
+
+    def verify_share_surveys(self, token, expected_status=200):
+        self.app.config['EMAIL_TOKEN_EXPIRY'] = 36000
+        response = self.client.get(f'/party-api/v1/share-survey/verification/{token}',
+                                   headers=self.auth_headers)
+        self.assertStatus(response, expected_status, "Response body is : " + response.get_data(as_text=True))
+        return json.loads(response.get_data(as_text=True))
```
Comment (warrenbailey) [inline test/party_client.py:375]: could this just be ?
```
response.getJson()
```

File: ras_party/controllers/share_survey_controller.py [hunk 2]
```diff
@@ -77,7 +85,95 @@ def get_unique_pending_shares(session):
...
+
+            except SQLAlchemyError as e:
+                logger.exception('Unable to confirm pending share survey', batch_no=batch_no)
+                session.rollback()
+                raise  # re raise the exception aimed at the generic handler
+        else:
+            logger.info('Ignoring respondent as already enrolled', business_id=business_id, survey_id=survey_id,
+                        email=share_surveys_list[0]['email_address'])
+        delete_share_survey_by_batch_no(batch_no, session)
+        session.commit()
```
Comment (warrenbailey) [inline ras_party/controllers/share_survey_controller.py]: i don't think you need this commit as its in the @with_db_session wrapper

File: ras_party/controllers/share_survey_controller.py [hunk 3]
```diff
@@ -77,7 +85,95 @@ def get_unique_pending_shares(session):
...
+            try:
+                with session.begin_nested():
+                    enrolment = Enrolment(business_respondent=business_respondent,
+                                          survey_id=pending_share_survey['survey_id'],
+                                          status=EnrolmentStatus.ENABLED)
+                    session.add(enrolment)
+
+            except SQLAlchemyError as e:
+                logger.exception('Unable to confirm pending share survey', batch_no=batch_no)
+                session.rollback()
```
Comment (warrenbailey) [inline ras_party/controllers/share_survey_controller.py]: again I think this is in the db wrapper

**Review by warrenbailey — CHANGES_REQUESTED** [review]
a couple of minor things but looks good

File: test/party_client.py (hunk 1 above)
Comment (creativeappdevamit) [reply test/party_client.py:375]: 👍

File: ras_party/controllers/share_survey_controller.py (hunk 2 above)
Comment (creativeappdevamit) [reply ras_party/controllers/share_survey_controller.py]: 👍

File: ras_party/controllers/share_survey_controller.py (hunk 3 above)
Comment (creativeappdevamit) [reply ras_party/controllers/share_survey_controller.py]: 👍

**Review by warrenbailey — APPROVED** [review]

**Pull request merged**